src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -221,54 +221,79 @@ private ImmutableArray<ImmutableArray<Project>> GetOrderedProjectsToProcess()
         {
             using var result = TemporaryArray<ImmutableArray<Project>>.Empty;
 
-            using var _ = PooledHashSet<Project>.GetInstance(out var processedProjects);
-
-            // First, if there's an active document, search that project first, prioritizing that active document and
-            // all visible documents from it.
-            if (_activeDocument != null)
+            // Get the initial set of project groups.  But filter out any projects that don't have an associated search
+            // service.  No point examining them or adding progress items for them.
+            foreach (var group in GetOrderedProjectsToProcessWorker())
             {
-                processedProjects.Add(_activeDocument.Project);
-                result.Add(ImmutableArray.Create(_activeDocument.Project));
-            }
+                using var groupCopy = TemporaryArray<Project>.Empty;
+                foreach (var project in group)
+                {
+                    if (_host.GetNavigateToSearchService(project) != null)
+                        groupCopy.Add(project);
+                }
 
-            // Next process all visible docs that were not from the active project.
-            using var buffer = TemporaryArray<Project>.Empty;
-            foreach (var doc in _visibleDocuments)
-            {
-                if (processedProjects.Add(doc.Project))
-                    buffer.Add(doc.Project);
+                if (groupCopy.Count > 0)
+                    result.Add(groupCopy.ToImmutableAndClear());
```

Comment:
Could this be simplified using WhereAsArray?

Reply:
yes.  done.